src/lines-yardstick.coffee

```diff
@@ -0,0 +1,164 @@
+    @headNode.appendChild(@stylesNode)
+    @headNode.appendChild(@syntaxStyleElement)
+
+    @contexts = []
+    @contexts.push(@createMeasurementContext()) for i in [0..8] by 1
+    @domNode.appendChild(context) for context in @contexts
+
+    @emitter.emit "did-initialize"
+
+  getRandomContextNode: ->
+    index = Math.round(Math.random() * 10)
+    index = Math.max(index, 0)
+    index = Math.min(index, @contexts.length - 1)
+
+    [index, @contexts[index]]
+
+  lineNodesForContextIndex: (contextIndex) ->
+    @lineNodesByContextIndex[contextIndex] ? []
+
+  buildDomNodesForScreenRows: (screenRows) ->
+    @ensureInitialized()
+
+    visibleLines = {}
+    html = ""
+    [contextIndex, contextNode] = @getRandomContextNode()
+
+    screenRows.forEach (screenRow) =>
+      line = @editor.tokenizedLineForScreenRow(screenRow)
+      visibleLines[line.id] = true
+
+      unless @screenRowsByLineId.hasOwnProperty(line.id)
+        lineState = @presenter.buildLineState(0, screenRow, line)
+        html += @linesBuilder.buildLineHTML(false, 1000, lineState)
+        @screenRowsByLineId[line.id] = screenRow
+
+    for lineNode in @lineNodesForContextIndex(contextIndex)
+      screenRow = lineNode.dataset.screenRow
+      lineId = @editor.tokenizedLineForScreenRow(screenRow)?.id
+
+      continue if visibleLines.hasOwnProperty(lineId)
+
+      delete @screenRowsByLineId[lineId]
+      lineNode.remove()
+
+    contextNode.insertAdjacentHTML("beforeend", html)
+
+    @lineNodesByContextIndex = {}
+    @lineNodesByScreenRow = {}
+    for domNode, i in @contexts when domNode isnt contextNode
+      @storeLineNodesInContextIndex(i)
+
+    @storeLineNodesInContextIndex(contextIndex)
+
+  storeLineNodesInContextIndex: (contextIndex) ->
+    contextNode = @contexts[contextIndex]
+
+    for lineNode in contextNode.getElementsByClassName("line")
+      screenRow = lineNode.dataset.screenRow
+      @lineNodesByScreenRow[screenRow] = lineNode
+      @lineNodesByContextIndex[contextIndex] ?= []
+      @lineNodesByContextIndex[contextIndex].push(lineNode)
+
+  lineNodeForScreenRow: (screenRow) ->
+    @lineNodesByScreenRow[screenRow]
+
+  leftPixelPositionForScreenPosition: (position) ->
+    @ensureInitialized()
+
+    lineNode = @lineNodeForScreenRow(position.row)
+
+    unless lineNode?
+      # console.log "#{position.row} not found. This wasn't expected."
+      return 0
+
+    tokens = lineNode.getElementsByClassName("token")
+
+    return 0 if tokens.length is 0
+
+    if foundToken = @findTokenByColumn(position.column, tokens)
+      textNode = foundToken.childNodes[0]
+      positionWithinToken = position.column - parseInt(foundToken.dataset.start)
+    else
+      textNode = last(tokens).childNodes[0]
+      positionWithinToken = textNode.textContent.length
+
+    @leftPixelPositionForCharInTextNode(textNode, positionWithinToken)
+
+  findTokenByColumn: (column, tokens, startIndex = 0, endIndex = tokens.length) ->
```

Comment:
To avoid finding a token through `TokenIterator` (which, by its nature, is sequential) I decorated the HTML tokens with some data attributes that contain screen position information. This allows us to perform a binary search on all the tokens in the current line.

The idea here is that we trade a few microseconds more on HTML parsing for a faster retrieval on the measurement side: while the former should be negligible, the latter brings a boost of `~ 2ms`.